cache/config.go

```diff
@@ -0,0 +1,41 @@
+package cache
+
+import "time"
+
+type Config struct {
+	ChannelSize    int
+	BufferEntries  int
+	DropThreshold  int
+	InitialBackoff time.Duration
+	MaxBackoff     time.Duration
+}
+
+func defaultConfig() Config {
+	return Config{
+		ChannelSize:    256,
```

Comment:
This is huge value to be assigned per watch, expect that there will be hundreds of watcher, allocating so much space would mean we allocate more memory per watch than is shared.